internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -781,12 +781,13 @@ func (c *Coordinator) DiagnosticHooks() diagnostics.Hooks {
 					State runtime.ComponentState `yaml:"state"`
 				}
 				type StateHookOutput struct {
-					State        agentclient.State      `yaml:"state"`
-					Message      string                 `yaml:"message"`
-					FleetState   agentclient.State      `yaml:"fleet_state"`
-					FleetMessage string                 `yaml:"fleet_message"`
-					LogLevel     logp.Level             `yaml:"log_level"`
-					Components   []StateComponentOutput `yaml:"components"`
+					State          agentclient.State      `yaml:"state"`
+					Message        string                 `yaml:"message"`
+					FleetState     agentclient.State      `yaml:"fleet_state"`
+					FleetMessage   string                 `yaml:"fleet_message"`
+					LogLevel       logp.Level             `yaml:"log_level"`
+					Components     []StateComponentOutput `yaml:"components"`
+					UpgradeDetails *details.Details       `yaml:"upgrade_details,omitempty"`
```

Comment:
I don't see any changes in the tests related to this change... could you please add a unit test to document what we can expect in the hook output when UpgradeDetails is populated ?

Reply:
Thanks.  Added in 02625e831564e5371bdaf6ca81771216f90df926.

Reply:
Thanks.  Added in 02625e831564e5371bdaf6ca81771216f90df926.